presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationPushDown.java

```diff
@@ -105,38 +106,36 @@ public void stopKafka()
     }
 
     @Test
-    public void testPartitionPushDown() throws ExecutionException, InterruptedException
+    public void testPartitionPushDown()
+            throws ExecutionException, InterruptedException
     {
         createMessages(topicNamePartition);
         String sql = format("SELECT count(*) FROM default.%s WHERE _partition_id=1", topicNamePartition);
 
         ResultWithQueryId<MaterializedResult> queryResult = getDistributedQueryRunner().executeWithQueryId(getSession(), sql);
-        assertEquals(getQueryInfo(getDistributedQueryRunner(), queryResult).getQueryStats().getProcessedInputPositions(), MESSAGE_NUM / 2);
+        assertEventually(() -> assertEquals(getQueryInfo(getDistributedQueryRunner(), queryResult).getQueryStats().getProcessedInputPositions(), MESSAGE_NUM / 2));
```

Comment:
This doesn't work. Final stats are set once. If they are wrong, they will stay wrong.

Reply:
oh... so we need to retry query execution as well.

Reply:
Unfortunately yes. We should fix the base issue, but that is some digging into complex scheduler.